common/src/types.rs

```diff
@@ -135,6 +136,27 @@ impl std::fmt::Debug for TxHash {
     }
 }
 
+impl Serialize for TxHash {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let b256: B256 = B256::from(self.as_array());
+        b256.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for TxHash {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let b256 = B256::deserialize(deserializer)?;
+        let array = <[u8; 32]>::try_from(b256.as_ref()).map_err(serde::de::Error::custom)?;
+        Ok(TxHash(array))
+    }
+}
+
```

Comment:
[`serde_as`](https://docs.rs/serde_with/latest/serde_with/attr.serde_as.html) или [`#[serde(from = ...)]`](https://serde.rs/container-attrs.html#from) . Второго хватит по идее.

P.S. `B256` должен уметь конвертиться в эррей без ошибок

P.P.S вообще только для десериализации тащить целый `reth_primitives` очень стремная идея. У нас уже есть `hex` в зависимостях, а у него есть [поддержка serde](https://docs.rs/hex/latest/hex/serde/index.html), лучше через нее сделать. А вообще странно, что этот тип не U256🤔

Reply:
`serde_as` requires additional dependency, and (correct me if I'm wrong) serde from/into requires a specific type, which in this case can be `String` but then we'll have to implement something like:

```rust
impl From<[u8; 32]> for TxHash {
    fn from(value: [u8; 32]) -> Self {
        Self(value)
    }
}

impl TryFrom<String> for TxHash {
    type Error = String;

    fn try_from(hex_string: String) -> Result<Self, Self::Error> {
        let bytes = hex_decode(&hex_string).map_err(|e| e.to_string())?;
        let array: [u8; 32] = bytes
            .try_into()
            .map_err(|_| "Expected a 32-byte array".to_string())?;
        Ok(TxHash(array))
    }
}
```

I'd rather leave Serialize/Deserialize implementation but using `hex` like you suggested instead of `B256`